content/300-guides/400-migrate-to-prisma/03-migrate-from-mongoose.mdx

```diff
@@ -0,0 +1,1066 @@
+  published    Boolean @default(false)
+  v            Int    @default(0) @map("__v") @ignore
+
+  author       User   @relation(fields: [authorId], references: [id])
+  authorId     String @map("author") @db.ObjectId
+
+  categories   Category[] @relation(fields: [categoryIds], references: [id])
+  categoryIds  String[]   @db.ObjectId
+
+  @@map("posts")
+}
+
+model User {
+  id      String        @id @default(auto()) @map("_id") @db.ObjectId
+  v       Int           @default(0) @map("__v") @ignore
+  email   String        @unique(map: "email_1")
+  name    String
+  profile Profile?
+  posts   Post[]
+
+  @@map("users")
+}
+```
+
+## Step 3. Install Prisma Client
+
+As a next step, you can install Prisma Client in your project so that you can start replacing the database queries in your project that are currently made with Mongoose:
+
+```terminal
+npm install @prisma/client
+```
+
+## Step 4. Replace your Mongoose queries with Prisma Client
+
+In this section, we'll show a few sample queries that are being migrated from Mongoose to Prisma Client based on the example routes from the sample REST API project. For a comprehensive overview of how the Prisma Client API differs from Mongoose, check out the [API comparison](/concepts/more/comparisons/prisma-and-mongoose#api-comparison) page.
+
+First, to set up the `PrismaClient` instance that you'll use to send database queries from the various route handlers. Create a new file named `prisma.js` in the `src` directory:
+
+```terminal copy
+touch src/prisma.js
+```
+
+Now, instantiate `PrismaClient` and export it from the file so you can use it in your route handlers later:
+
+```js copy file=src/prisma.js
+const { PrismaClient } = require('@prisma/client')
+
+const prisma = new PrismaClient()
+
+module.exports = prisma
+```
+
+The imports in our controller files are as follows:
+
+```js file=src/controllers/post.js
+const Post = require('../models/post')
+const User = require('../models/user')
+const Category = require('../models/category')
+```
+
+```js file=src/controllers/user.js
+const Post = require('../models/post')
+const User = require('../models/user')
+```
+
+You'll update the controller imports as you migrate from Mongoose to Prisma:
+
+```js file=src/controllers/post.js
+const prisma = require('../prisma')
+```
+
+```js file=src/controllers/user.js
+const prisma = require('../prisma')
+```
+
+### 4.1. Replacing queries in `GET` requests
+
+The REST API has four routes that accept `GET` requests:
+
+- `/feed?searchString={searchString}&take={take}&skip={skip}`: Return all published posts
+  - Query Parameters (optional):
+    - `searchString`: Filter posts by `title` or `content`
+    - `take`: Specifies how many objects should be returned in the list
+    - `skip`: Specifies how many of the returned objects should be skipped
+- `/post/:id`: Returns a specific post
+- `/authors`: Returns a list of authors
+
+Let's dive into the route handlers that implement these requests.
+
+#### `/feed`
```

Comment:
```suggestion
#### The `/feed` handler
```

Reply:
Alex, the heading/title should be more than just the name of the handler... I leave it up to you if you want to add "The" and "handler" to each heading. You could try it and see how it looks.